drafts/FS-1146-scoped-nowarn.md

```diff
@@ -0,0 +1,224 @@
+# F# RFC FS-1146-scoped-nowarn
+
+The design suggestion [278](https://github.com/fsharp/fslang-suggestions/issues/278) has been marked "approved in principle".
+
+This RFC covers the detailed proposal for this suggestion.
+
+- [x] [Suggestion](https://github.com/fsharp/fslang-suggestions/issues/278)
+- [x] Approved in principle
+- [ ] [Implementation](https://github.com/dotnet/fsharp/pull/18049)
+- [X] [Discussion](https://github.com/fsharp/fslang-design/discussions/786)
+
+# Summary
+
+Allow the #nowarn directive to become scoped (per warning number) by a corresponding #warnon directive.
+
+# Motivation
+
+Today, the #nowarn directive disables the specified warnings for all code from the directive to the end of the file. Usually, however, people want to see all warnings except for some very specific situations. Therefore, a way to disable warnings for some fragment of the code is a much-requested feature.
+
+Quotes form the language suggestion thread:
+
+- "being able to nowarn a few lines is most critical for my workplace"
+- "it would be huge if we could get this"
+- "it would be great to do only small sections that we know we want to ignore rather than the whole file"
+
+Motivating examples, again from the suggestion thread:
+
+```
+// We know f is never called with a None.
+let f (Some a) =    // creates warning 25, which we want to suppress
+    // 2000 loc, where the incomplete match warning is beneficial
+ ```
+
+```
+// We have a union with an obsolete case
+type MyErrors =
+| CoolError1 of CoolError1
+| CoolError2 of CoolError2
+| [<Obsolete>] OldError1 of OldError1
+
+let handler e =
+    match e with
+    | CoolError1 ce -> ...
+    | CoolError2 ce -> ...
+    | OldError ce -> ...   // warning 44, which we want to suppress only here
+```
+
+# Detailed specification
+
+1. The compiler shall recognize a new *compiler directive* `#warnon` (to be added to §12.4 of the F# spec).
```

Comment:
Apart from the spec and from https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/compiler-directives#preprocessor-directives , will any other doc pages need updating?

Reply:
The last section of the RFC mentions all the places that I found. 
This includes a small entry in the formatting guidelines.